grpc/binlog/v1alpha/binarylog.proto

```diff
@@ -52,15 +52,36 @@ message GrpcLogEntry {
   // The logger uses one of the following fields to record the payload,
   // according to the type of the log entry.
   oneof payload {
-    // Used by CLIENT_INIT_METADATA, SERVER_INIT_METADATA and TRAILING_METADATA
+    // Used by {SEND,RECV}_INITIAL_METADATA and
+    // {SEND,RECV}_TRAILING_METADATA. This contains only the metadata
+    // from the application.
     Metadata metadata = 4;
-    // Used by REQUEST and RESPONSE
+    // Used by {SEND,RECV}_MESSAGE
     Message message = 5;
   }
 
   // Peer address information, will only be recorded in SEND_INITIAL_METADATA
   // and RECV_INITIAL_METADATA entries.
   Peer peer = 6;
+
+  // true if payload does not represent the full message or metadata.
+  bool truncated = 7;
+
+  // The method name. Always logged for every entry.
```

Comment:
Why every entry?  Seems like this will waste a lot of space, since there will generally be at least 6 entries per call (and arbitrarily many for streaming).

Reply:
Why would a log message get dropped?  And if it does, since the information for the call will be incomplete in the log anyway, why do we care if the remaining info can be decoded?

Reply:
The underlying writer implementation that we use is allowed to drop writes when experiencing memory pressure. I also notice that it does not guarantee consistent ordering across all CPUs. It would be nice to still offer some info in the degraded mode, but there's no requirement to do this. 

Let's only log for the initial message, and not put too much effort into incompletely logged RPCs. I'm not sure how much of a problem they actually are.

Reply:
Once https://github.com/grpc/grpc-proto/pull/19 is sorted out, the final entry for the rpc (whether it is a normal termination or a cancellation) should probably log a count of how many entries are expected for this RPC.